  fix another incompatibility of generator3 with PyQt4  